server/api/impl/admin/install.go

```diff
@@ -48,7 +48,7 @@ func (adm *Admin) InstallApp(cc *apps.Context, sessionToken apps.SessionToken, i
 	client := model.NewAPIv4Client(conf.MattermostSiteURL)
 	client.SetToken(string(sessionToken))
 
-	if app.GrantedPermissions.Contains(apps.PermissionActAsBot) && app.BotUserID == "" {
```

Comment:
I thought this was protecting from double-registering the bot account and the OAuth app? Or were these checks redundant? OT: We need an E2E test for Install/Uninstall. It's time.

Reply:
The part that takes care of the double-registering is the "ensure" function. With the previous approach you were considering that the previous value was always an existing app, and will not create a new one. In that case, if the admin deleted by mistake the oauth app, and wanted to fix it reinstallin, it will not work, because the app id (or bot id) exist in the app, even if it does not exist in the system.

Reply:
> I thought this was protecting from double-registering the bot account and the OAuth app? Or were these checks redundant? OT: We need an E2E test for Install/Uninstall. It's time.

cc: @iomodo